From ea9cc01b7fe38971ef943aa88c0d5d00dfb03ebb Mon Sep 17 00:00:00 2001
From: Junping Du <junping_du@apache.org>
Date: Tue, 5 May 2015 10:18:23 -0700
Subject: [PATCH 2129/2848] YARN-3396. Handle URISyntaxException in
 ResourceLocalizationService. (Contributed by
 Brahma Reddy Battula) (cherry picked from commit
 38102420621308f5ba91cdeb6a18a63aa5acf640)

(cherry picked from commit 6ea007ba91bbe460de7509351562ee06b3f77533)

Conflicts:
	hadoop-yarn-project/CHANGES.txt

Change-Id: I9c07c3248ee0b08fc9474f4c4a3dbae163f73ba3
---
 .../localizer/ResourceLocalizationService.java     |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/ResourceLocalizationService.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/ResourceLocalizationService.java
index 127420a..bb779fa 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/ResourceLocalizationService.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/localizer/ResourceLocalizationService.java
@@ -1008,7 +1008,9 @@ LocalizerHeartbeatResponse processHeartbeat(
         try {
           req = new LocalResourceRequest(rsrc);
         } catch (URISyntaxException e) {
-          // TODO fail? Already translated several times...
+          LOG.error(
+              "Got exception in parsing URL of LocalResource:"
+                  + rsrc.getResource(), e);
         }
         LocalizerResourceRequestEvent assoc = scheduled.get(req);
         if (assoc == null) {
@@ -1082,7 +1084,9 @@ LocalizerHeartbeatResponse processHeartbeat(
           LOG.error("Inorrect path for PRIVATE localization."
               + next.getResource().getFile(), e);
         } catch (URISyntaxException e) {
-            //TODO fail? Already translated several times...
+          LOG.error(
+              "Got exception in parsing URL of LocalResource:"
+                  + next.getResource(), e);
         }
       }
 
-- 
1.7.9.5

